google/cloud/storage/retry.py

```diff
@@ -35,19 +39,34 @@
     requests.ConnectionError,
     requests_exceptions.ChunkedEncodingError,
     requests_exceptions.Timeout,
+    http.client.BadStatusLine,
```

Comment:
LGTM, thanks for consolidating the retryable errors. Looking at the requests docs `requests.ConnectionError` should be equivalent to `requests.exceptions.ConnectionError` so we're good!